mpmt_afterpulse.cpp

```diff
@@ -0,0 +1,201 @@
+                  
+      }
+
+    }
+  
+  }
+
+  // std::cout<<histQlsr->GetXaxis()->GetBinCenter(27)<<std::endl;
+  // std::cout<<histQlsr->GetBinContent(27)<<std::endl;
+
+  // estimating afterpulse rate.
+  // as afp per lsr, and afp per total if necessary.
+  afpAverageLsr1   = double(afpCounter1)/double(lsrCounter1);
+  afpAverageLsr2   = double(afpCounter2)/double(lsrCounter2);
+  afpAverageLsr3   = double(afpCounter3)/double(lsrCounter3);
+  afpAverageLsr4   = double(afpCounter4)/double(lsrCounter4);
+  afpAverageLsr5   = double(afpCounter5)/double(lsrCounter5);
+  afpAverageLsr6   = double(afpCounter6)/double(lsrCounter6);
+  // std::cout<<"Laser pulses: " << lsrCounter << std::endl;
+  // std::cout<<"Afterpulses: " << afpCounter << std::endl;
+  std::cout<< afpAverageLsr1*100.<< " percent." << std::endl;
+  std::cout<< afpAverageLsr2*100.<< " percent." << std::endl;
+  std::cout<< afpAverageLsr3*100.<< " percent." << std::endl;
+  std::cout<< afpAverageLsr4*100.<< " percent." << std::endl;
+  std::cout<< afpAverageLsr5*100.<< " percent." << std::endl;
+  std::cout<< afpAverageLsr6*100.<< " percent." << std::endl;
+
+
+  std::cout<<"lsrCounter1: "<<lsrCounter1<<std::endl;
+  std::cout<<"lsrCounter2: "<<lsrCounter2<<std::endl;
+  std::cout<<"lsrCounter3: "<<lsrCounter3<<std::endl;
+  std::cout<<"lsrCounter4: "<<lsrCounter4<<std::endl;
+  std::cout<<"lsrCounter5: "<<lsrCounter5<<std::endl;
+  std::cout<<"lsrCounter6: "<<lsrCounter6<<std::endl;
+
+  std::cout<<"afpCounter1: "<<afpCounter1<<std::endl;
+  std::cout<<"afpCounter2: "<<afpCounter2<<std::endl;
+  std::cout<<"afpCounter3: "<<afpCounter3<<std::endl;
+  std::cout<<"afpCounter4: "<<afpCounter4<<std::endl;
+  std::cout<<"afpCounter5: "<<afpCounter5<<std::endl;
+  std::cout<<"afpCounter6: "<<afpCounter6<<std::endl;
+
+  // TIME histogram
+  canvas1->cd();
+  histT->SetTitle("Pulse times");
+  histT->GetXaxis()->SetTitle("Time (ns)");
+  histT->GetYaxis()->SetTitle("No. of events");
+  histT->GetYaxis()->SetRange(0,100);
+  histT->Draw();
+  canvas1->SaveAs(fnmT);
+
+  // ALL PULSES histogram
+  canvas2->cd();
+  histQ->SetTitle("Total pulses, charge");
+  histQ->GetXaxis()->SetTitle("Charge");
+  histQ->GetYaxis()->SetTitle("No. of events");
+  histQ->GetXaxis()->SetRange(0,100);
+  histQ->Draw();
+  canvas2->SaveAs(fnmQ);
+
+  // LASER PULSES histogram
+  canvas3->cd();
+  histQlsr->SetTitle("Laser pulses, charge");
+  histQlsr->GetXaxis()->SetTitle("Charge");
+  histQlsr->GetYaxis()->SetTitle("No. of events");
+  histQlsr->GetXaxis()->SetRange(0,100);
+  histQlsr->Draw();
+  canvas3->SaveAs(fnmQlsr);  
+  canvas3->SaveAs("../Laser charges.png");  
+
+  // AFTER PULSES histogram
+  canvas4->cd();
+  histQafp->SetTitle("Afterpulses, charge");
+  histQafp->GetXaxis()->SetTitle("Charge");
+  histQafp->GetYaxis()->SetTitle("No. of events");
+  histQafp->GetXaxis()->SetRange(0,100);
+  histQafp->Draw();
+  canvas4->SaveAs(fnmQafp);
+  return 0;
+}
```

Comment:
Script to create the afterpulse histograms. Counting the number of after pulse based on threshold value.We calculate the after pulse for 6PMT?